.claude/scripts/agent-telemetry.sh

```
@@ -749,6 +749,25 @@ newest_first() {
+# Root transcripts only, capped AFTER the subagent filter. Filtering downstream
+# of `head -n MAX_FILES` lets sidechains evict eligible root runs before the
+# classifier ever sees them — with subagents routinely outnumbering roots here,
+# the published dispatch count could reach zero while root runs existed.
+root_session_files() {
+  [ -d "$CLAUDE_PROJECTS" ] || return 0
+  if store_root_in_scope; then
+    newest_first "$CLAUDE_PROJECTS" | sort -rn | cut -d' ' -f2- \
+      | grep -v '/subagents/' | head -n "$MAX_FILES"
+    return 0
+  fi
+  ls -1 "$CLAUDE_PROJECTS" 2>/dev/null \
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)